src/gui/dialogs/unit_recall.cpp

```diff
@@ -235,6 +240,36 @@ void unit_recall::pre_show(window& window)
 	list_item_clicked(window);
 }
 
+void unit_recall::rename_unit(window& window)
+{
+	listbox& list = find_widget<listbox>(&window, "recall_list", false);
+
+	const int index = list.get_selected_row();
+	const unit& selected_unit = *recall_list_[index].get();
+
+	std::string name = selected_unit.name();
+	const std::string dialog_title(N_("Rename Unit"));
+	const std::string dialog_label(N_("Name:"));
+
+	if(gui2::dialogs::edit_text::execute(dialog_title, dialog_label, name)) {
+		unit_ptr unit_to_rename = team_.recall_list().find_if_matches_id(selected_unit.id());
+		assert(unit_to_rename);
+		unit_to_rename->rename(name);
+
+		find_widget<label>(list.get_row_grid(index), "unit_name", false).set_label(name);
+
+		filter_options_.erase(filter_options_.begin() + index);
+		std::string filter_text = unit_to_rename->type_name() + " " + name + " " + std::to_string(unit_to_rename->level());
```

Comment:
Please use an `std::ostringstream` here and in the loop right below instead of a plain `std::string`.